tools.go

```diff
@@ -84,31 +88,90 @@ func ConvertTool[T any, R any](name, description string, toolHandler ToolHandler
 	}
 
 	handler := func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+		// Create OpenTelemetry span for tool execution (safely, with fallback)
+		var span trace.Span
+		
+		// Only create spans if tracing is explicitly enabled
```

Comment:
I wonder if these checks are actually needed 🤔 we should be able to create spans anyway, and they just won't be propagated anywhere if tracing is disabled? it would make this code a lot cleaner.